### PR TITLE
Feat: Designed UI for 404:Error Page for default redirects.

### LINKED ISSUE #115: Feat: Design UI for 404:Error Page for default redirects.
## Is your feature request related to a problem? Please describe.

Currently, there is no feature to redirect the user to a specific Error 404 Page when they enter the wrong URLs. There should be a design and change in Route paths for this.

## Describe the solution you'd like

- Design a 404 Error Page

- Route redirect to implement this.

## Describe alternatives you've considered
  N/A

## Additional context
Would like to work on this.

### PR DESCRIPTION
**Pull request by Aaishpra**
### Description

Created an ErrorPage for default redirects on wrong or broken urls

- [x] Design a 404 Error Page

- [x] Route redirect to implement this.

Fixes #115 

### Type of Change:

- Code
- Quality Assurance
- User Interface

**Code/Quality Assurance Only**
- Bugfix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Screenshot for reference
image

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

- [x] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by Aaishpra** [comment]
@mtreacy002 have a look.

**Review by mtreacy002 — CHANGES_REQUESTED** [review]
@Aaishpra . The 404 error page looks good from the layout/design perspective. However, I think we have an issue here. With this PR, this 404 error page is returned when user navigate to any other pages that is not under `Home` tab. This is not a desired behaviour. What should happen is for any of the urls that exist in the `Route`, the intended page should be returned as normal. Only when the url is not known/given in the `Route`, the app is expected to return the 404 page.

* bug when user click on any of the protected pages as well as Register and Login, 404 error page is returned
Screen Shot 2020-10-24 at 10 40 52 pm

* an example of the accepted event when 404 should be shown (e.g. when useer click url that is not given in the `Route`)
Screen Shot 2020-10-24 at 10 53 52 pm

Can you please fix this? Thanks

**Comment by Aaishpra** [comment]
@mtreacy002 i tried solving the bug by replacing the 

`<Route path=""><ErrorPage /></Route>`

with the following commands

`<Route path="*"><ErrorPage /></Route>`

`<Route><ErrorPage /></Route>`

`<Route component = {ErrorPage} />`

but none of it seems to work 
so I decided to use ProtectedRoute instead of Route and got the following error
image
In my view this has to do something with the following code in ProtectedRoute 
image
Can you tell me how can i solve this ?

File: src/Routes.js [hunk 1]
```diff
@@ -38,6 +39,9 @@ export default function Routes() {
           <Route exact path="/benefit">
             <Benefit />
           </Route>
+          <Route path = "">
```
Comment (mtreacy002) [inline src/Routes.js]: Hi @Aaishpra . I found a solution for this. It's just the matter of placing the code of the error page route within the switch options. It works fine once I removed the path and placed this error page route at the very bottom of the switch options. You can use this method on your code.

Screen Shot 2020-10-28 at 11 44 50 am

ezgif com-gif-maker (16)

**Review by mtreacy002 — CHANGES_REQUESTED** [review]
@Aaishpra . I found a solution for your issue. Please check the comment below and modify your code accordingly.

**Comment by Aaishpra** [comment]
@mtreacy002 thanks for the solution,  it seems to work properly now.

File: src/Routes.js (hunk 1 above)
Comment (Aaishpra) [reply src/Routes.js]: Done

**Review by mtreacy002 — APPROVED** [review]
This PR is tested locally on MacOS Catalina 10.15.7 with Chrome browser.
The following is the test result:
* expected behaviour: all existing pages with known urls are rendered properly. Only unknown url displaying error page 404 not found
* actual behaviour: as expected
    * when user not logged-in
ezgif com-gif-maker (17)

    * when user logged-in
ezgif com-gif-maker (18)

Well done, @Aaishpra and thank you for your contribution 🎉🎉🎉

**Pull request merged**